dsgrid/dataset/table_format_handler_base.py

```diff
@@ -65,22 +68,19 @@ def add_columns(
                     "Adding time columns through supplemental mappings is not supported yet."
                 )
             records = self._project_config.get_base_to_supplemental_mapping_records(query_name)
-            if not aggregation_allowed:
+            if aggregation_columns is None:
                 to_ids = records.groupBy("from_id").agg(F.count("to_id").alias("count_to_id"))
                 counts_of_to_id = get_unique_values(to_ids, "count_to_id")
                 if counts_of_to_id != {1}:
                     raise DSGInvalidParameter(
                         f"Mapping dimension query name {query_name} produced duplicate to_ids for one or more from_ids"
                     )
-            from_fractions = get_unique_values(records, "from_fraction")
-            if len(from_fractions) != 1 and float(next(iter(from_fractions))) != 1.0:
-                # TODO #199: This needs to apply from_fraction to each load value column.
-                # Also needs to handle all possible from_id/to_id combinations
-                # If aggregation is not allowed then it should raise an error.
-                raise DSGInvalidParameter(
-                    f"Mapping dimension query name {query_name} produced from_fractions other than 1.0: {from_fractions}"
-                )
-            records = records.drop("from_fraction")
+                from_fractions = get_unique_values(records, "from_fraction")
+                if len(from_fractions) != 1 and float(next(iter(from_fractions))) != 1.0:
+                    raise DSGInvalidParameter(
+                        f"Mapping dimension query name {query_name} produced from_fractions other than 1.0: {from_fractions}"
+                    )
+
```

Comment:
Correct. However, I've been thinking about this case and have mostly concluded that it should be removed. We support the notion of `supplemental_columns` in the query result data model. It means that you can add a column, such as `state` to the table without having to fill out the aggregation section (which is tedious). That is duplicate to specifying `geography: ["county", "state"]` in the aggregation section.

1. Having two ways of doing the same thing is confusing to users.
2. The current implementation has a bug with ordering if you specify both an aggregation, such as county to state and a supplemental column, such as census_region. The best solution is to make the user specify all the columns they want in the aggregation section so that we can apply it in one groupBy expression.

Do you see any reason not to remove the notion of supplemental columns?

Reply:
Yes--I think we wanted people to be able to see the county column with the state column also available, because counties are easier to understand if the state information is handy.

Reply:
I just realized that this feature could be really helpful for the crazy TEMPO subsectors--if we made a TEMPO-specific subsector class then we could have the mashed up ids AND each element broken out in a separate column.

Reply:
Your reference to a new subsector class makes me think that we are missing each other. The `supplemental_columns` feature uses dimension mappings, not columns in the dimension record file.

Consider this example.
- User specifies that they want an aggregation of county to state and inserts `geography: ["state"]`.
- User wants the census region in the result table and so adds `supplemental_columns: ["census_region"]`

The supplemental columns have to be added before the base dimension is dropped. Then the code has to track these new columns during the aggregation process (there is currently a bug which can be fixed.)

My thought yesterday that this is all too complex for the users and the code and that we should give them one place to specify what columns they want in the table. The downside is that it is more complex to type in the JSON file. I think that's why we added the feature in the first place.

Let's go back to the case of just adding the state column (no aggregation).

Current recommendation:
```
supplemental_columns: ["state"]
```

Proposed:
```
aggregations: [
  {
    aggregation_function: "sum",
    dimensions: {
       geography: ["county", "state"]
       # every other dimension type + query name
    }
  }
]
```
There is a CLI command that auto-generates the aggregation section with base dimension query names, but it produces the more verbose form:
```
  "geography": [
    {
      "dimension_query_name": "county",
      "function": null,
      "alias": null
    }
  ],
```

Reply:
The `supplemental_columns` feature is removed.